doc/langdef.md

```diff
@@ -116,44 +126,185 @@ In general it is a bad idea for those defining contexts or extensions to use
+double-quotes, and may contain any unescaped characters except for the delimiter
+sequence.  Again, the closing delimiter must match the opening one. Triple-quoted
+strings may contain newlines.
+
+Both sorts of strings can include escape sequences, described below.
+
+If preceded by an `r` or `R` character, the string is a _raw_ string and does
+not interpret escape sequences. Raw strings are useful for expressing strings
+which themselves must use escape sequences, such as regular expressions or
+program text.
+
+Bytes literals are represented by string literals preceded by a `b` or `B`
+character. The bytes literal is the sequence of bytes given by the UTF-8
+representation of the string literal. In addition, the octal escape sequence are
+interpreted as octet values rather than as Unicode code points. Both raw and
+multiline string literals can be used for byte literals.
+
+Escape sequences are a backslash (`\`) followed by one of the following:
+
+*   A single-quote(`'`), double-quote(`"`), or backslash(`\`), representing
+    itself.
+*   A code for whitespace:
+    *   `b`: backspace
+    *   `f`: form feed
+    *   `n`: line feed
+    *   `r`: carriage return
+    *   `t`: horizontal tab
+*   A `u` followed by four hexadecimal characters, encoding a Unicode code point
+    in the BMP. Characters in other Unicode planes can be represented with
+    surrogate pairs.
+*   Three octal digits, in the range `000` to `377`. For strings, it denotes the
+    unicode code point. For bytes, it represents an octet value.
+
+Examples:
+
+CEL Literal  | Meaning
+------------ | --------------------------------------------------------------
+`""`         | Empty string
+`'""'`       | String of two double-quote characters
+`'''x''x'''` | String of four characters "x''x"
+`"\""`       | String of one double-quote character
+`"\\"`       | String of one backslash character
+`r"\\"`      | String of two backslash characters
+`b"abc"`     | Byte sequence of 61, 62, 63
+`b"&yuml;"`  | Sequence of bytes 204 and 191 (UTF-8 of &yuml;)
+`b"\303\277"`| Also sequence of bytes 204 and 191
+`"\303\277"` | String of "&Igrave;&iquest;" (code points 204, 191)
+`"\377"`     | String of "&yuml;" (code point 255)
+`b"\377"`    | Sequence of byte 255 (_not_ UTF-8 of &yuml;)
+
+### Aggregate Values
+
+Lists are ordered sequences of values.
+
+Maps are a set of key values, and a mapping from these keys to arbitrary values.
+Key values must be an allowed key type: `int`, `uint`, `bool`, or `string`.
+
+Note that the type checker uses a finer-grained notion of list and map types.
+Lists are `list(A)` for the homogenous type `A` of list elements. Maps are
+`map(K, V)` for maps with keys of type `K` and values of type `V`.  The type
+`dyn` is used for heterogeneous values  See
+[Gradual Type Checking](#gradual-type-checking).
+
+Any protocol buffer message is a CEL value, and each message type is its own CEL
+type, represented as its name (e.g. `google.protobuf.timestamp`).
```

Comment:
When referring to the timestamp message name, use `Timestamp`
rather than `timestamp`

Reply:
Done, here and elsewhere.